katharsis-client/src/main/java/io/katharsis/client/KatharsisClient.java

```diff
@@ -0,0 +1,226 @@
+
+	private ResourceRegistry resourceRegistry;
+
+	private ModuleRegistry moduleRegistry;
+	private RequestUrlBuilder urlBuilder;
+
+	private boolean initialized = false;
+
+	private ExceptionMapperRegistry exceptionMapperRegistry;
+
+	/**
+	 * @param serviceUrl
+	 * @param resourcePackageName
+	 */
+	public KatharsisClient(String serviceUrl, String resourceSearchPackage) {
+		if (serviceUrl.endsWith("/")) {
+			serviceUrl = serviceUrl.substring(0, serviceUrl.length() - 1);
+		}
+
+		resourceRegistry = new ResourceRegistry(serviceUrl);
+		urlBuilder = new RequestUrlBuilder(resourceRegistry);
+
+		objectMapper = new ObjectMapper();
+		objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+		moduleRegistry = new ModuleRegistry();
+		moduleRegistry.addModule(new CoreModule(resourceSearchPackage, new ResourceFieldNameTransformer()));
+
+		// TODO add module support
+		JsonApiModuleBuilder moduleBuilder = new JsonApiModuleBuilder();
+		SimpleModule jsonApiModule = moduleBuilder.build(resourceRegistry, true);
+		jsonApiModule.addDeserializer(BaseResponseContext.class,
+				new BaseResponseDeserializer(resourceRegistry, objectMapper));
+		jsonApiModule.addDeserializer(ErrorResponse.class, new ErrorResponseDeserializer());
+
+		// TODO add to official module
+		SimpleModule clientModule = new SimpleModule();
+
+		objectMapper.registerModule(jsonApiModule);
+		objectMapper.registerModule(clientModule);
+	}
+
+	protected void init() {
+		if (initialized)
+			return;
+		initialized = true;
+
+		initModuleRegistry();
+		initRepositories();
+		initExceptionMapperRegistry();
+	}
+
+	private void initModuleRegistry() {
+		moduleRegistry.init(objectMapper, resourceRegistry);
+	}
+
+	private void initRepositories() {
+		// register all resources
+		ResourceLookup resourceLookup = moduleRegistry.getResourceLookup();
+		Set<Class<?>> resourceClasses = resourceLookup.getResourceClasses();
+		for (Class<?> resourceClass : resourceClasses) {
+			allocateRepository(resourceClass);
+		}
+	}
+
+	private void initExceptionMapperRegistry() {
+		ExceptionMapperLookup exceptionMapperLookup = moduleRegistry.getExceptionMapperLookup();
+		exceptionMapperRegistry = new ExceptionMapperRegistryBuilder().build(exceptionMapperLookup);
+	}
+
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private <T, ID extends Serializable> void allocateRepository(Class<T> resourceClass) {
+		ResourceInformation resourceInformation = moduleRegistry.getResourceInformationBuilder().build(resourceClass);
+		final ResourceRepositoryStub<T, ID> repositoryStub = new ResourceRepositoryStubImpl<T, ID>(this, resourceClass,
+				resourceInformation, urlBuilder);
+
+		// create interface for it!
+		RepositoryInstanceBuilder repositoryInstanceBuilder = new RepositoryInstanceBuilder(null, null) {
+			@Override
+			public Object buildRepository() {
+				return repositoryStub;
+			}
+		};
+		ResourceEntry<T, ID> resourceEntry = new DirectResponseResourceEntry<T, ID>(repositoryInstanceBuilder);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)